unix/xserver/hw/vnc/meson.build

```diff
@@ -0,0 +1,115 @@
+tigervnc_srcdir = join_paths(meson.source_root(), '../..')
+tigervnc_builddir = tigervnc_srcdir
+
+cpp = meson.get_compiler('cpp')
+
+vnccommon_deps = [
+    declare_dependency(
+        dependencies: cpp.find_library('rfb', dirs: join_paths(tigervnc_builddir, 'common/rfb'))),
+    declare_dependency(
+        dependencies: cpp.find_library('rdr', dirs: join_paths(tigervnc_builddir, 'common/rdr'))),
+    declare_dependency(
+        dependencies: cpp.find_library('os', dirs: join_paths(tigervnc_builddir, 'common/os'))),
+    declare_dependency(
+        dependencies: cpp.find_library('network', dirs: join_paths(tigervnc_builddir, 'common/network'))),
+    declare_dependency(
+        dependencies: cpp.find_library('unixcommon', dirs: join_paths(tigervnc_builddir, 'unix/common'))),
```

Comment:
I'm afraid I'm unfamiliar with meson. Will this still use our `.la` files? Or do we need to come up with some new glue for meson?

It is important that it not only finds the `.a` files, but also includes any dependent libraries. Possibly also cflags, but I'm unsure how well that even works currently.

Reply:
Hmm, I don't think meson is capable of handle .la files.

This first attempt manually adds the dependencies, as you can see (e.g. `dependency('zlib')`).

Now that I see, this works only on my build setup. A workaround could be to do the dynamic library selection to build the common libs at hw/vnc/meson.build. 

If the common libs were built with meson too, the library dependencies could be known.

Reply:
That duplication is not something we'll reasonably be able to maintain.

Moving everything from CMake to meson seems overkill. So we want to add some glue, like we have right now between CMake and libtool.

Please have a look at `CMakeMacroLibtoolFile.cmake` and see how we can do something similar for meson. I.e. automatically generate dependency information in a format meson can consume.